### PR TITLE
[SofaRigid] Clean JointSpringFF

### PR DESCRIPTION
**Pull request by VannesteFelix**
Cleaning of JointSpringFF according to SOFA conventions 
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

File: SofaKernel/modules/SofaRigid/JointSpringForceField.inl [hunk 1]
```diff
@@ -42,75 +35,77 @@ namespace component
 namespace interactionforcefield
 {
 
-static const double pi=3.14159265358979323846264338327950288;
+using sofa::helper::vector;
```
Comment (guparan) [inline SofaKernel/modules/SofaRigid/JointSpringForceField.inl]: We said we would avoid this `using` during last dev meeting ;-)

**Review by epernod — COMMENTED** [review]
If I get it you move the definition of a spring inside a class and use it inside JointSpringFF ?
Do you think this can be factorized with other springFF?

**Review by epernod — CHANGES_REQUESTED** [review]
Also could you check the CI, several tests are broken:
https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-850/CI_CONFIG=ubuntu_gcc-5.4,CI_PLUGINS=options,CI_TYPE=release/2/warnings4Result/

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by epernod** [comment]
**https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-850/CI_CONFIG=ubuntu_gcc-5.4,CI_PLUGINS=options,CI_TYPE=release/2/**  warnings4Result/package.-377079774/

\+ 

see BUILD_URL/**artifact/parent_dir/build/scene-tests/examples/Components/constraint/FixedLMConstraint.scn/output.txt**

=

https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-850/CI_CONFIG=ubuntu_gcc-5.4,CI_PLUGINS=options,CI_TYPE=release/2//artifact/parent_dir/build/scene-tests/examples/Components/forcefield/JointSpringForceField.scn/output.txt

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

File: SofaKernel/modules/SofaRigid/JointSpring.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace component
+{
+
+namespace interactionforcefield
+{
+
+using namespace sofa::defaulttype;
+
+#ifndef SOFA_FLOAT
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaRigid/JointSpring.cpp]: Nooooo ! The Sofa float are trying to re-enter in Sofa :)

File: SofaKernel/modules/SofaRigid/JointSpring.h [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        if (s.initRot[3]!= 1)
+            out<<"REST_R "<<s.initRot<<"  ";
+
+        out<<"END_SPRING"<<std::endl;
+        return out;
+}
+};
+
+#if  !defined(SOFA_JOINTSPRING_CPP)
+#ifndef SOFA_FLOAT
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaRigid/JointSpring.h]: To remove.

File: SofaKernel/modules/SofaRigid/JointSpring.h [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                Real softKst = 0, Real hardKst = 10000, Real softKsr = 0, Real hardKsr = 10000, Real blocKsr = 100,
+                Real axmin = -100000, Real axmax = 100000, Real aymin = -100000, Real aymax = 100000, Real azmin = -100000, Real azmax= 100000,
+                Real kd = 0);
+
+    /// Attributes
+    int  m1, m2;                    /// the two extremities of the spring: masses m1 and m2
+    Real kd;                        /// damping factor
+    Vector torsion;                 /// torsion of the springs in axis/angle format
+    Vector lawfulTorsion;           /// projected torsion in allowed angles
+    Vector KT;                      /// linear stiffness
```
Comment (guparan) [inline SofaKernel/modules/SofaRigid/JointSpring.h]: All these inline comments (above and below) need `///<` to be correctly parsed by Doxygen :wink:

**Review by guparan — CHANGES_REQUESTED** [review]

File: SofaKernel/modules/SofaRigid/JointSpring.cpp (hunk 2 above)
Comment (VannesteFelix) [reply SofaKernel/modules/SofaRigid/JointSpring.cpp]: ooops... fixed

File: SofaKernel/modules/SofaRigid/JointSpring.h (hunk 3 above)
Comment (VannesteFelix) [reply SofaKernel/modules/SofaRigid/JointSpring.h]: fixed

File: SofaKernel/modules/SofaRigid/JointSpring.h (hunk 4 above)
Comment (VannesteFelix) [reply SofaKernel/modules/SofaRigid/JointSpring.h]: I didn't know, thanks.
Fixed now

**Comment by VannesteFelix** [comment]
@guparan & @epernod  is ok for you now ?

**Comment by damienmarchal** [comment]
Did this one ready ? What is missing ?

**Comment by marques-bruno** [comment]
I think it's just that it's never been changed back to to_review after @VannesteFelix's changes.. 
We didn't discuss it this morning at the dev meeting.
@guparan @epernod?

**Comment by guparan** [comment]
Ready!

**Review by guparan — APPROVED** [review]

**Pull request merged**